app/vmauth/main.go

```diff
@@ -350,3 +350,14 @@ func handleConcurrencyLimitError(w http.ResponseWriter, r *http.Request, err err
 	}
 	httpserver.Errorf(w, r, "%s", err)
 }
+
+func isNetworkRetryableError(err error) bool {
```

Comment:
I offer an alternative version:

```go
func isNetworkRetryableError(err error) bool {
	// The syscall.ECONNREFUSED doesn't work on Windows.
	// Windows will return WSAECONNREFUSED(syscall.Errno = 10061) when the connection is refused.
	if errors.Is(err, syscall.Errno(10061)) ||
		errors.Is(err, syscall.ECONNREFUSED) ||
		errors.Is(err, syscall.ECONNRESET) ||
		errors.Is(err, syscall.ENETUNREACH) ||
		errors.Is(err, syscall.ECONNABORTED) ||
		errors.Is(err, syscall.EHOSTDOWN) ||
		errors.Is(err, syscall.EHOSTUNREACH) {
		return true
	}

	var dnsError *net.DNSError
	if errors.As(err, &dnsError) {
		return dnsError.IsNotFound || dnsError.IsTemporary
	}

	if errors.Is(err, io.EOF) || errors.Is(err, io.ErrUnexpectedEOF) {
		return true
	}

	if os.IsTimeout(err) || errors.Is(err, syscall.ETIMEDOUT) {
		return true
	}

	return false
}
```

But it's better to discuss it in a voice tomorrow :)

Reply:
syscalls're platform specific. It may fail to compile at some targets and requires build constrains. 

 Also, we already have function with string match logic for error handling: `netutil.IsTrivialNetworkError`.
@valyala  WDYT?

Reply:
Changed logic.